### DISCUSSION TITLE
First launching issue

### DISCUSSION THREAD

**Discussion by nek-to** [skiptools/skip#220] [Q&A] [answered]
Hey, everybody! Need some help.
I have a problem with a project build.
I have **Xcode**, **Android studio** and **brew** installed.
I install java **jdk v17** through oracle site.
The skip checkup command works **without errors**, **only with one warning** about Rosetta.
I create a project via `skip init --appid=bundle.id project-name AppName`
After loading and indexing I try to build the project, but when building in Xcode I get an error 
```
ERROR: JAVA_HOME is set to an invalid directory: /opt/homebrew/opt/java

Please set the JAVA_HOME variable in your environment to match the
location of your Java installation.
```

I set the PATH to JAVA_HOME via echo $JAVA_HOME. 
`/usr/local/Cellar/openjdk@17/17.0.12/libexec/openjdk.jdk/Contents/Home`
Changing it to` /opt/homebrew/opt/java` doesn't change anything.

I will be glad to any hints!
Thanks!

**Comment by marcprux** [comment, marked as answer]
The Rosetta error likely indicates that you have an old Intel installation of Homebrew. The best solution is to remove and re-install Homebrew from https://brew.sh, and then re-install Skip. That should place the correct Java (which, BTW, needs to be Java 22 and not 17) in `/opt/homebrew/opt/java`.

If this still fails, can you post the output of the `skip checkup` command, along with attaching the log file if it fails?

**Reply by nek-to** [reply]
Thx for your answer!
It didn't work...
What I did:
1) Removed homebrew  `/bin/bash -c “$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/uninstall.sh)”`
2) Cleaned up `sudo rm -rf /opt/homebrew`
3) Installed homebrew `/bin/bash -c “$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)”`
4) Installed java v22 use `brew install openjdk@22`
5) Installed skip use` brew install skiptools/skip/skip`
6) Performed a brew `skip checkup` with the following result
[✓] Skip version 1.0.10 (= 1.0.10)
[✓] macOS version 14.6.1 (> 13.5.0)
[!] macOS architecture: wrong architecture 1, Rosetta must not be enabled for process on (Intel)
[✓] Swift version 5.10 (> 5.9.0)
[✓] Xcode version 15.4 (> 15.0.0)
[✓] Xcode tools SDKs: 5
[✓] Homebrew version 4.3.19 (> 4.1.0)
[✓] Gradle version 8.10 (> 8.6.0)
[✓] Java version 22.0.2 (> 21.0.0)
[⢇] Android Studio version[✓] Android Debug Bridge version 1.0.41 (> 1.0.40)
[✓] Android Studio version: 2024.1
[✓] Android tools SDKs: 2
[✓] Resolve dependencies (42.74s)
[✓] Build hello-skip (69.33s)
[✓] Test Swift (33.3s)
[✓] Test Kotlin (154.17s)
[✓] Archive iOS ipa (43.19s)
[✓] Assemble HelloSkip-release.ipa 38 KB
[✓] Verify HelloSkip-release.ipa 38 KB
[✓] Assembling Android apk (296.05s)
[✓] Verify HelloSkip-release.apk 13,1 MB
[✓] Check Swift Package (1.19s)
[✓] Check Skip Updates: 1.0.10
[!] Skip 1.0.10 checkup with 1 warning
7) Created a project use `skip init --open-xcode --appid=com.xyz.HelloSkip hello-skip HelloSkip`
8) When I tried to build again I got the error
```
 ERROR: JAVA_HOME is set to an invalid directory: /opt/homebrew/opt/java

Please set the JAVA_HOME variable in your environment to match the
location of your Java installation.
```

<img width="795" alt="Снимок экрана 2024-09-02 в 19 57 15" src="https://github.com/user-attachments/assets/fe651133-9296-4301-9d06-5685ef3cb0ca">

**Reply by marcprux** [reply]
Can you run `which java` and `which skip` and let us know what it says, as well as `echo $JAVA_HOME`? If it is in `/usr/local/`, then it is still using the old Intel version. Can you check your ~/.zshrc file and ensure that your PATH does not contain `/usr/local/`?

**Reply by nek-to** [reply]
You were right!
The problem was that I was running iTerm with Rosetta, thus not running in an arm64 environment.
I turned off Rosetta in the iTerm properties and repeated all the steps above.
I was able to build the project!)
Thx for your help

**Comment by youpelegrace** [comment]
Hello here, i have same error mine is 

ERROR: JAVA_HOME is set to an invalid directory: /opt/homebrew/Cellar/openjdk@17/17.0.12/libexec/openjdk.jdk/Contents/Home

Please set the JAVA_HOME variable in your environment to match the
location of your Java installation.

i have tried a couple of things including installing and installing new java and also setting the java path to the correct one but anytime i try to run the app, i get same error.  Running Android was working perfectly till i updated my Mac OS to the latest and this started
